### PR TITLE
Login and Logout tests updated

### LINKED ISSUE #5: Update test script for Authentication
- Update this https://github.com/nerdyfactory/nf-react-app-template/blob/master/src/App.test.tsx to use mock API prepared in this https://github.com/nerdyfactory/nf-react-app-template/issues/2

### PR DESCRIPTION
**Pull request by nikolasvasconcelos**
fix #5

### CODE REVIEW

**Review by JonghunYu — CHANGES_REQUESTED** [review]
@nikolasvasconcelos

Currently, you triggered the `login` function directly in the test script.
But, please add a form in the login screen add an event handler in the component and fill them and click button in the script so that we can replicate the user's behavior

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...atedRules":"30"},{"filePath":"53","messages":"54","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/index.tsx",[],[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/reportWebVitals.ts",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/App.tsx",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/navigations/index.ts",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/contexts/index.ts",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/components/index.ts",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/contexts/AuthContext.ts",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/navigations/PrivateNav.tsx",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/navigations/PublicNav.tsx",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/components/Loading/index.tsx",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/pages/index.ts",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/pages/Logout/index.tsx",[],"/Users/nikolasvasconcelos/Work/nf-react-app-template/src/pages/Login/index.tsx",[]]
```
Comment (JonghunYu) [inline .eslintcache]: Please gitignore this file

File: src/__tests__/App.test.tsx [hunk 2]
```diff
@@ -1,38 +1,68 @@
...
-      const { container } = loginAfterRender();
+      const { container } = await loginAfterRender();
       expect(container).toMatchSnapshot();
     });
 
     it('moves to login screen', async () => {
-      loginAfterRender();
+      await loginAfterRender();
       fireEvent.click(screen.getByText('Logout'));
+      await logout();
```
Comment (JonghunYu) [inline src/__tests__/App.test.tsx]: I think `fireEvent.click(screen.getByText('Logout'));` triggers logout under the hood so we don't need to explicitly call `logout` function in the test

File: src/pages/Login/index.tsx [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
 import { AuthContext, AuthActionTypes } from '../../contexts';
 
 export function Login() {
   const { dispatch } = useContext(AuthContext);
-  const onLogin = () => dispatch({ type: AuthActionTypes.SET_TOKEN, payload: 'somerandometoken' });
+  const [user, setUser] = useState<string>(``);
+  const [password, setPassword] = useState<string>(``);
+
+  const onLogin = () => {
+    // login(user, password);
```
Comment (JonghunYu) [inline src/pages/Login/index.tsx]: I think this should call login function

**Review by JonghunYu — CHANGES_REQUESTED** [review]
@nikolasvasconcelos I think you didn't implement the original `login` and `logout` functions to be mocked by src/__mocks__/user.ts
Why do we need a mocked function of the functions that don't exist?

**Comment by nikolasvasconcelos** [comment]
@JonghunYu Now I understood what you expected. I actually thought about letting the UI itself trigger the functions but I did not find a proper way to make it work, since those tutorials teach how to mock API only in the test file itself, and my experience with testing isn't much. I'll focus on finishing this ASAP.

**Comment by nikolasvasconcelos** [comment]
@JonghunYu removed login and logout functions from testing scripts and moved to an auth hook, which is called by the user interface

File: .eslintcache (hunk 1 above)
Comment (JonghunYu) [inline .eslintcache]: This file shouldn't be here. Please delete this https://stackoverflow.com/questions/2047465/how-can-i-delete-a-file-from-a-git-repository

File: src/__mocks__/user.ts [hunk 4]
```diff
@@ -1,6 +1,20 @@
-import request, { User } from './request';
+import { USER_TOKEN } from 'constants/utility';
```
Comment (JonghunYu) [inline src/__mocks__/user.ts]: Remove this file completely

File: src/__tests__/Api.test.ts [hunk 5]
```diff
@@ -1,10 +1,10 @@
-import * as user from '../__mocks__/user';
+import { getUserName } from '../__mocks__/user';
```
Comment (JonghunYu) [inline src/__tests__/Api.test.ts]: I don't think we need this test file anymore

File: src/hooks/useAuth.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { AuthActionTypes, AuthContext } from 'contexts';
+import { useContext } from 'react';
+import { JWT_FAKE_TOKEN } from '__mocks__/request';
+
+const useAuth = () => {
+  const { dispatch } = useContext(AuthContext);
+
+  const login = async (username: string, password: string) => {
+    // DEVELOPER'S IMPLEMENTATION
```
Comment (JonghunYu) [inline src/hooks/useAuth.ts]: Add implementation here that calling `singIn` and use token from response

File: src/hooks/useAuth.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    console.log(username, password);
+    dispatch({ type: AuthActionTypes.SET_TOKEN, payload: `${JWT_FAKE_TOKEN}` });
+  };
+
+  const logout = async () => {
+    // DEVELOPER'S IMPLEMENTATION
```
Comment (JonghunYu) [inline src/hooks/useAuth.ts]: I don't think we need additional implementation or mock for the logout. Removing token from `localStorage` is enough

**Review by JonghunYu — CHANGES_REQUESTED** [review]
@nikolasvasconcelos Please check the comments below

**Comment by JonghunYu** [comment]
@nikolasvasconcelos Please think why it tries to mock request in this example https://jestjs.io/docs/en/tutorial-async

This example has `getUserName` function calling `request` inside which is mocked.
In our case, we have `login` function and it will call another function returning token and store it in the local storage and we should mock the function. not `login` itself

**Review by JonghunYu — APPROVED** [review]
@nikolasvasconcelos It looks good now. Thanks

**Pull request merged**